src/anndata/utils.py

```diff
@@ -400,3 +400,12 @@ def is_hidden(attr) -> bool:
             for item in type.__dir__(cls)
             if not is_hidden(getattr(cls, item, None))
         ]
+
+
+def raise_value_error_if_multiindex_columns(df: pd.DataFrame):
+    if isinstance(df.columns, pd.MultiIndex):
+        msg = (
+            "MultiIndex columns are not supported in AnnData. "
+            "Please use a single-level index."
```

Comment:
Please add (and use) an attr name parameter here. This should make it easier if the error gets raised during initialization.

```suggestion
            f"Please use a single-level index for {attr}."
```